backend/main.go

```diff
@@ -125,17 +199,39 @@ func Run(cmd *cobra.Command, args []string) error {
 
 	operationsScanner := NewOperationsScanner(dbClient, ocmConnection)
 
-	stop := make(chan struct{})
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	signalChannel := make(chan os.Signal, 1)
 	signal.Notify(signalChannel, syscall.SIGINT, syscall.SIGTERM)
-
-	go operationsScanner.Run(logger, stop)
-
-	sig := <-signalChannel
-	logger.Info(fmt.Sprintf("caught %s signal", sig))
-	close(stop)
-
-	operationsScanner.Join()
+	go func() {
+		sig := <-signalChannel
```

Comment:
Please just use `signal.NotifyContext` on this thing? Didn't we already talk about this?

Reply:
I want to log the signal received along with a timestamp so I can observe how much longer it takes the process to shut down.  I don't see any big advantage to `signal.NotifyContext`; it's a convenience wrapper that does the same thing I'm doing.

Reply:
Is `signalChannel` just for your logger? Then the following is simpler:

```go
ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
defer cancel()

go func() {
    <-ctx.Done()
    logger.Info("Caught interrupt signal")
}()
```

And the only difference is you can't tell if you got SIGINT or SIGTERM (why do you care? in production, the kubelet only ever sends SIGTERM or SIGKILL ([ref](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)), so this seems like a distinction without much meaning). Code is much less Rube-Goldberg. IDK. I respect you think this logging of the specific signal that is sent is somehow important but in a decade of writing k8s-native applications it has never been useful or required in the past.

Reply:
As far as big advantage, I did want to mention - using a more complex cascade of handlers and goroutines when it's not necessary is a downside for sure. Making the simple choice every time you can helps make this maintainable ten years down the line.